src/main/java/io/liftbridge/Message.java

```diff
@@ -25,93 +27,101 @@ static Message fromWire(Api.Message wireMsg) {
         return msg;
     }
 
-    Api.Message toWire() {
-        Api.Message.Builder msg = Api.Message.newBuilder();
-        return msg.build();
-    }
-
-	/**
-	 * @return the offset
-	 */
-	public Long getOffset() {
-		return offset;
-	}
-	/**
-	 * @param offset the offset to set
-	 */
-	private void setOffset(Long offset) {
-		this.offset = offset;
-	}
-	/**
-	 * @return the key
-	 */
-	public byte[] getKey() {
-		return key;
-	}
-	/**
-	 * @param key the key to set
-	 */
-	private void setKey(byte[] key) {
-		this.key = key;
-	}
-	/**
-	 * @return the value
-	 */
-	public byte[] getValue() {
-		return value;
-	}
-	/**
-	 * @param value the value to set
-	 */
-	private void setValue(byte[] value) {
-		this.value = value;
-	}
-	/**
-	 * @return the timestampNanos
-	 */
-	public Long getTimestampNanos() {
-		return timestampNanos;
-	}
-	/**
-	 * @param timestampNanos the timestampNanos to set
-	 */
-	private void setTimestampNanos(Long timestampNanos) {
-		this.timestampNanos = timestampNanos;
-	}
-	/**
-	 * @return the subject
-	 */
-	public String getSubject() {
-		return subject;
-	}
-	/**
-	 * @param subject the subject to set
-	 */
-	private void setSubject(String subject) {
-		this.subject = subject;
-	}
-	/**
-	 * @return the replySubject
-	 */
-	public String getReplySubject() {
-		return replySubject;
-	}
-	/**
-	 * @param replySubject the replySubject to set
-	 */
-	private void setReplySubject(String replySubject) {
-		this.replySubject = replySubject;
-	}
-	/**
-	 * @return the headers
-	 */
-	public HashMap<String, byte[]> getHeaders() {
-		return headers;
-	}
-	/**
-	 * @param headers the headers to set
-	 */
-	private void setHeaders(HashMap<String, byte[]> headers) {
-		this.headers = headers;
-	}
+    /**
+     * @return the offset
+     */
+    public Long getOffset() {
+        return offset;
+    }
+
+    /**
+     * @param offset the offset to set
+     */
+    private void setOffset(Long offset) {
+        this.offset = offset;
+    }
+
+    /**
+     * @return the key
+     */
+    public byte[] getKey() {
+        return key;
+    }
+
+    /**
+     * @param key the key to set
+     */
+    private void setKey(byte[] key) {
+        this.key = key;
+    }
+
+    /**
+     * @return the value
+     */
+    public byte[] getValue() {
+        return value;
+    }
+
+    /**
+     * @param value the value to set
+     */
+    private void setValue(byte[] value) {
+        this.value = value;
+    }
+
+    /**
+     * @return the timestampNanos
+     */
+    public Long getTimestampNanos() {
```

Comment:
One question here would be whether it makes sense to even expose the timestamp as nanos. Nanos is used simply because that's how Go implements time, but it might make more sense to expose a more idiomatic time in the Java client?

Reply:
We can use time types instead too. I opted for using the nanos to be closer to the API, but it might be a good idea to have a more idiomatic type. Just gotta make sure it's not going to lose the nanos precision

Reply:
I created an issue to track this